src/types/user.d.ts

```diff
@@ -1,8 +1,20 @@
-interface User {
+type User = {
 	email: string;
 	username: string;
 	passwordHash: string;
 	avatarUrl?: string | null;
 	createdAt: string;
 	userId: string;
-}
\ No newline at end of file
+};
+
+type GoogleUser = Omit<User, "passwordHash"> & { googleId: number };
+
+type GoogleUserInfo = {
+	sub: number;
+	name: string;
+	given_name: string;
+	family_name: string;
+	picture: string; // URL
+	email: string;
+	email_verified: true;
+};
```

Comment:
A badass move would be to use inheritance here to shine on your haters.
```

interface BaseUser {
    email: string;
    username: string;
    createdAt: string;
    userId: string;
    avatarUrl?: string | null;
}

interface User extends BaseUser {
    passwordHash: string;
}

interface GoogleUserInfo {
    sub: number;
    name: string;
    given_name: string;
    family_name: string;
    picture: string; // URL
    email: string;
    email_verified: true;
}

type GoogleUser = Omit<BaseUser, "avatarUrl"> & { googleId: number };
```
this way it would be easier to add, for example, an apple login or a facebook login.